From abc619af0d8a4060c6a9aeee3de53e51eac73b6b Mon Sep 17 00:00:00 2001
From: test <test@test.com>
Date: Thu, 21 May 2026 09:07:56 +0000
Subject: [PATCH] dylib-lit-paths


diff --git a/test/lit.cfg.py b/test/lit.cfg.py
index 727c07d..fb2cd5a 100644
--- a/test/lit.cfg.py
+++ b/test/lit.cfg.py
@@ -99,3 +99,31 @@ if config.libfst_enabled:
   config.available_features.add('libfst')
 
 llvm_config.add_tool_substitutions(tools, tool_dirs)
+
+
+def find_shlibpath_var():
+    if platform.system() in ["Linux", "FreeBSD", "NetBSD", "OpenBSD", "SunOS"]:
+        yield "LD_LIBRARY_PATH"
+    elif platform.system() == "Darwin":
+        yield "DYLD_LIBRARY_PATH"
+    elif platform.system() == "Windows":
+        yield "PATH"
+    elif platform.system() == "AIX":
+        yield "LIBPATH"
+
+
+for shlibpath_var in find_shlibpath_var():
+    # in stand-alone builds, shlibdir is clang's build tree
+    # while llvm_libs_dir is installed LLVM (and possibly older clang)
+    shlibpath = os.path.pathsep.join(
+        (
+            config.circt_shlib_dir,
+            config.environment.get(shlibpath_var, ""),
+        )
+    )
+    config.environment[shlibpath_var] = shlibpath
+    break
+else:
+    lit_config.warning(
+        "unable to inject shared library path on '{}'".format(platform.system())
+    )
