From 3c1b4b73cabe4d441af7d3f6872ae89adb4a2fab Mon Sep 17 00:00:00 2001
From: test <test@test.com>
Date: Thu, 21 May 2026 09:08:21 +0000
Subject: [PATCH] mlir-tblgen: fix


diff --git a/CMakeLists.txt b/CMakeLists.txt
index 764681e..8a3bd67 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,6 +65,11 @@ if( CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR )
   set(CIRCT_BUILT_STANDALONE 1)
   set(BACKEND_PACKAGE_STRING "LLVM ${LLVM_PACKAGE_VERSION}")
 
+  # MLIR doesn't export this target, nor any variable specifying its binary dir.
+  # Look through LLVM's binary directory as a hint.
+  find_program(MLIR_TABLEGEN_EXE "mlir-tblgen" HINTS ${LLVM_TOOLS_BINARY_DIR}
+    NO_DEFAULT_PATH)
+
   # Handle unittests when building out-of-tree against an installed version of
   # LLVM/MLIR (not a build tree). Adapted from `llvm/flang/CMakeLists.txt`.
   set(CIRCT_GTEST_AVAILABLE 0)
@@ -124,7 +129,6 @@ else()
   set(MLIR_MAIN_SRC_DIR ${LLVM_MAIN_SRC_DIR}/../mlir ) # --src-root
   set(MLIR_INCLUDE_DIR ${MLIR_MAIN_SRC_DIR}/include ) # --includedir
   set(MLIR_TABLEGEN_OUTPUT_DIR ${LLVM_BINARY_DIR}/tools/mlir/include)
-  set(MLIR_TABLEGEN_EXE $<TARGET_FILE:mlir-tblgen>)
   # Keep the in-tree MLIR headers ahead of dependency prefixes that might also
   # contain an installed MLIR checkout.
   include_directories(BEFORE "${MLIR_INCLUDE_DIR}")
